File: frontend/src/types/session.ts
export enum SessionStatus {
  CREATED = 'CREATED',
  CONNECTING = 'CONNECTING',
  ACTIVE = 'ACTIVE',
  DISCONNECTED = 'DISCONNECTED',
  RECONNECTING = 'RECONNECTING',
  IDLE = 'IDLE',
  ENDED = 'ENDED'
}

export interface Session {
  id: string
  user_id: string
  title: string | null
  status: SessionStatus
  created_at: string
  last_activity: string
  ended_at: string | null
}

export interface SessionCreate {
  title?: string
  working_directory?: string
}

File: backend/src/schemas/session.py
from pydantic import BaseModel, Field, ConfigDict
from datetime import datetime
import uuid
from ..models.session import SessionStatus

class SessionCreate(BaseModel):
    title: str | None = Field(None, max_length=255)
    working_directory: str = Field(default=".")

class SessionUpdate(BaseModel):
    title: str | None = Field(None, max_length=255)

class SessionResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    title: str | None
    status: SessionStatus
    created_at: datetime
    last_activity: datetime
    ended_at: datetime | None

    model_config = ConfigDict(from_attributes=True)
